File: code/stable/projectile/src/python/InputConstraints.py
## \file InputConstraints.py
# \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
# \brief Provides the function for checking the physical constraints and software constraints on the input
from __future__ import print_function
import sys
import math

import InputParameters

## \brief Verifies that input values satisfy the physical constraints and software constraints
# \param inParams structure holding the input values
def input_constraints(inParams):
    if (not(inParams.v_launch > 0)) :
        print("Warning: ", end='')
        print("v_launch has value ", end='')
        print(inParams.v_launch, end='')
        print(" but suggested to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")
    if (not(0 < inParams.theta and inParams.theta < 3.14159265 / 2)) :
        print("Warning: ", end='')
        print("theta has value ", end='')
        print(inParams.theta, end='')
        print(" but suggested to be ", end='')
        print("between ", end='')
        print(0, end='')
        print(" and ", end='')
        print(3.14159265 / 2, end='')
        print(" ((pi)/(2))", end='')
        print(".")
    if (not(inParams.p_target > 0)) :
        print("Warning: ", end='')
        print("p_target has value ", end='')
        print(inParams.p_target, end='')
        print(" but suggested to be ", end='')
        print("above ", end='')
        print(0, end='')
        print(".")

File: code/stable/projectile/src/csharp/InputConstraints.cs
/** \file InputConstraints.cs
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Provides the function for checking the physical constraints and software constraints on the input
*/
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

public class InputConstraints {
    
    /** \brief Verifies that input values satisfy the physical constraints and software constraints
        \param inParams structure holding the input values
    */
    public static void input_constraints(InputParameters inParams) {
        if (!(inParams.v_launch > 0)) {
            Console.Write("Warning: ");
            Console.Write("v_launch has value ");
            Console.Write(inParams.v_launch);
            Console.Write(" but suggested to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
        }
        if (!(0 < inParams.theta && inParams.theta < 3.14159265 / 2)) {
            Console.Write("Warning: ");
            Console.Write("theta has value ");
            Console.Write(inParams.theta);
            Console.Write(" but suggested to be ");
            Console.Write("between ");
            Console.Write(0);
            Console.Write(" and ");
            Console.Write(3.14159265 / 2);
            Console.Write(" ((pi)/(2))");
            Console.WriteLine(".");
        }
        if (!(inParams.p_target > 0)) {
            Console.Write("Warning: ");
            Console.Write("p_target has value ");
            Console.Write(inParams.p_target);
            Console.Write(" but suggested to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
        }
    }
}

